13-Tenant Security Solution Setup/Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1

```diff
@@ -0,0 +1,639 @@
+    Write-Host "`n" 
+
+    Write-Host "Validating whether the current user [$($currentSub.Account.Id)] have the required permissions to run the script for Subscription [$($SubscriptionId)]..."
+    Write-Host "`n"
+
+    # Safe Check: Checking whether the current account is of type User
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor $([Constants]::MessageType.Warning)
+        break;
+    }
+    
+    Write-Host "Fetching storage account from Subscription [$($SubscriptionId)]..."
+    
+    # Array to store resource context
+    $resourceContext = @()
+    $controlIds = "Azure_Storage_AuthN_Dont_Allow_Anonymous"
+    
+    # If json path not given fetch all storage account.
+    if([string]::IsNullOrWhiteSpace($Path))
+    {
+        $resourceContext = Get-AzStorageAccount
+    }
+    else
+    {
+        if (-not (Test-Path -Path $Path))
+        {
+            Write-Host "Error: Json file containing storage account(s) detail not found for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            break;        
+        }
+
+        # Fetching storage accounts details for remediation.
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.FailedControlSet
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId};
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -and ($resourceDetails.ResourceDetails | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No storage account(s) found in input json file for remedition." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+        $resourceDetails.ResourceDetails | ForEach-Object { 
+            try
+            {
+                $resourceContext += Get-AzStorageAccount -Name $_.StorageAccountName -ResourceGroupName $_.ResourceGroupName
+            }
+            catch
+            {
+                Write-Host "Valid resource group and resource name not found in input json file. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                break
+            }
+        }
+    }
+
+    $totalStorageAccount = ($resourceContext | Measure-Object).Count
+    
+    switch ($RemediationType)
+    {
+        "DisableAllowBlobPublicAccessOnStorage"
+        {
+            try
+            {
+                if($totalStorageAccount -gt 0)
+                {
+                    $stgWithEnableAllowBlobPublicAccess = @()
+                    $stgWithDisableAllowBlobPublicAccess = @()
+                    $resourceContext | ForEach-Object {
+                        $stg = Get-AzStorageAccount -ResourceGroupName $_.ResourceGroupName -StorageAccountName $_.StorageAccountName
```

Comment:
$resourceContext already has output of Get-AzStorageAccount, do we need to call Get-AzStorageAccount again here?

Reply:
Updated script, no need to call 'Get-AzStorageAccount' command, we have all required data to check further condition.